core/dbt/events/functions.py

```diff
@@ -12,11 +11,14 @@
 import sys
 from typing import Callable, Dict, Optional, TextIO
 import uuid
+from google.protobuf.json_format import MessageToDict
 
 
 LOG_VERSION = 3
 metadata_vars: Optional[Dict[str, str]] = None
 
+nofile_codes = ["Z012", "Z013", "Z014", "Z015"]
```

Comment:
I added a comment. In theory we could still use something like classes or method flags, but we'd have to get the class from msg.info.name, do getattr(types, class_name) and then check for the existence of another class or execute a class method. So that's still on option if some other flag thing comes along that makes it worth it.

Reply:
cool! agree that, in this case, it doesn't feel worth it — hard-coding the codes is a right-sized solution